### PR TITLE
New verification

### PR DESCRIPTION
**Pull request by todinov**
Will squash before merge. It's easier to review by commits.

### CODE REVIEW

**Comment by Foriger** [comment]
Can you squash the commits.

File: mpinsdk/src/main/java/com/miracl/mpinsdk/MPinMfaAsync.java [hunk 1]
```diff
@@ -305,6 +306,58 @@ public void run() {
...
+     * @param clientId
+     *   The client ID associated with the user
+     * @param redirectUri
+     *   The redirectUri...
+     * @param accessCode
+     *   A valid access code
+     * @param callback
+     *   Callback for the operation
+     */
+    public void startVerification(final @NonNull User user, final @NonNull String clientId, final @NonNull String redirectUri, final @NonNull String accessCode, @NonNull final Callback<Void> callback){
```
Comment (Foriger) [inline mpinsdk/src/main/java/com/miracl/mpinsdk/MPinMfaAsync.java]: redirectURI could be a more general type like URL.

File: README.md [hunk 2]
```diff
@@ -129,19 +130,19 @@ The returned value is a newly created `User` instance. The User class itself loo
...
 ```java
 public class User implements Closeable {
- 
+
     public enum State {
         INVALID,
         STARTED_REGISTRATION,
         ACTIVATED,
         REGISTERED,
         BLOCKED
```
[the remark this answers is not in the document]
Comment (todinov) [reply README.md]: Fixed

File: mpinsdk/src/main/java/com/miracl/mpinsdk/MPinMfaAsync.java (hunk 1 above)
Comment (todinov) [reply mpinsdk/src/main/java/com/miracl/mpinsdk/MPinMfaAsync.java]: I don't think we should change this at the moment.

File: README.md [hunk 3]
```diff
@@ -268,14 +294,11 @@ The Platform will also start a user identity verification procedure, by sending
...
- 
+
 Optionally, the application might pass additional `pushToken`, which is a unique token for sending _Push Notifications_ to the mobile app.
 When such token is provided, the Platform might use additional verification step by sending a Push Notification to the app.
 
-Additional optional parameter is the `regCode`.
-This is a _Registration Code_ that could be used to bypass the identity verification process.
-A valid registration code could be generated by an already registered device, after authenticating the user.
-This code could then be provided during the registration process on a device, and the Platform will let the user register, skipping the verification process for that identity.
+The `regCode` is a code which value indicates that the user identity is already verified. It could be obtained as `activationToken` value of `VerificationResult` object from a successfull call to `finishVerification()` method using the default identity verification or using a bootstrap code.
```
Comment (tatiNo5) [inline README.md]: Can you make the method `finishVerification` a link to its description?

File: README.md [hunk 4]
```diff
@@ -256,10 +258,34 @@ When the mobile app has the Authorization URL, it can pass it to this method as
...
+The `VerificationResult` class returned as a reference variable has the following form:
+
+```
+public class VerificationResult {
+    public String accessCode;
+    public String activationToken;
+}
+```
+
+The `accessCode` is a session identifier which you could control the session with by `abortSession()`, `getSessionDetails()`.
```
Comment (tatiNo5) [inline README.md]: Can you make the methods links to their description?

File: README.md (hunk 3 above)
Comment (tatiNo5) [reply README.md]: Will be done in another PR. We need consistency with the rest of the readme :)

File: README.md (hunk 4 above)
Comment (tatiNo5) [reply README.md]: Will be done in another PR. We need consistency with the rest of the readme :)

**Pull request merged**